source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/admin.js

```diff
@@ -70,12 +78,63 @@
             CurrentPosition: this.currentPosition,
             Seniority: this.seniority,
             FullName: this.fullName,
-            Period: this.period
+            Period: this.period,
+            IsValid: this.isValid()
         };
     }
-
     var viewmodel = new PeriodCreation();
 
+    $('#generate-evaluation-button').on('click', function () {
+        if (viewmodel.isValid()) {
+            postEvaluationsForGeneration({
+                onSuccess: function (result) {
+                    showEvaluationGenerationResult(true, result.amountGenerated);
+                },
+                onError: function () {
+                    alert('Fallo interno. Por favor recargue la página.');
+                }
+            });
+        } else {
+            showEvaluationGenerationResult(false);
+        }
+
+        function postEvaluationsForGeneration(callbacks) {
+            var model = viewmodel.toJs();
+            var modelFiltered = { Employees: _.filter(model.Employees, function (e) { return e.ResponsibleId && !e.Period; }) };
+            $.ajax("/Evaluations/api/GenerateEvalutions/" + evaluationPeriod + "/", {
+                type: "POST",
+                dataType: 'json',
+                contentType: 'application/json; charset=utf-8',
+                data: JSON.stringify(modelFiltered),
+                complete: function (response) {
+                    var result = { amountGenerated: parseInt(response.responseText) }
+                    if (typeof callbacks.onSuccess === 'function') { callbacks.onSuccess(result); }
+                }
+            });
+            error = callbacks.onError;
+        };
+
+        function showEvaluationGenerationResult(wereGenerated, amountGenerated) {
```

Comment:
No, the point is that each function/routine should have a single purpose. The event handler is that: handling the event and calling functionalities from other functions.

We didn't have that before and you broke it into pieces (which is perfect), which each piece having it's own single purpose.

My concern right here is that everything is still inside the handler, and it should be declared outside of the handler. The way that you separated the logic is fine, but you should take it out of the handler.